app/models/grda_warehouse/tasks/service_history/enrollment.rb

```diff
@@ -21,13 +21,23 @@ def self.batch_process_unprocessed!(max_wait_seconds: 21_600)
         wait_for_processing(max_wait_seconds: max_wait_seconds)
     end
 
+    scope :unassigned, -> do
+      jobs = Delayed::Job.where(queue: :long_running, failed_at: nil).
+        jobs_for_class('ServiceHistory::RebuildEnrollments').
+        pluck(:id)
```

Comment:
Doesn't this need to pluck the ids out of the handler, not the delayed job ids?

Reply:
So, this will work, but I wonder if it makes more sense to not store the delayed_job_id and just look at the handler to see which enrollments are already queued....